therapi-runtime-javadoc/src/test/java/com/github/therapi/runtimejavadoc/internal/parser/CommentParserTest.java

```diff
@@ -0,0 +1,110 @@
+    public void parse_textOnly(String text) {
+        List<CommentElement> elements = CommentParser.parse(text).getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new CommentText(text), elements.get(0));
+    }
+
+    @Test
+    public void parse_linkOnly_simpleLink() {
+        List<CommentElement> elements = CommentParser.parse("{@link ClassName}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineLink(new Link("ClassName", "ClassName", null)), elements.get(0));
+    }
+
+    @Test
+    public void parse_linkOnly_labeledLink_noWhiteSpace() {
+        List<CommentElement> elements = CommentParser.parse("{@link ClassName myLabel}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineLink(new Link("myLabel", "ClassName", null)), elements.get(0));
+    }
+
+    @Test
+    public void parse_linkOnly_labeledLink_withWhiteSpace() {
+        List<CommentElement> elements = CommentParser.parse("{@link ClassName my label}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineLink(new Link("my label", "ClassName", null)), elements.get(0));
+    }
+
+    @Test
+    public void parse_linkOnly_linkWithMemberRef() {
+        List<CommentElement> elements = CommentParser.parse("{@link ClassName#member}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineLink(new Link("ClassName#member", "ClassName", "member")), elements.get(0));
+    }
+
+    @Test
+    public void parse_linkOnly_labeledLinkWithMemberRef() {
+        List<CommentElement> elements = CommentParser.parse("{@link ClassName#member label}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineLink(new Link("label", "ClassName", "member")), elements.get(0));
+    }
+
+    @Test
+    public void parse_tagOnly_noWhiteSpace() {
+        List<CommentElement> elements = CommentParser.parse("{@sometag someValue}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineTag("sometag", "someValue"), elements.get(0));
+    }
+
+    @Test
+    public void parse_tagOnly_withWhiteSpace() {
+        List<CommentElement> elements = CommentParser.parse("{@sometag some value}").getElements();
+        assertEquals(1, elements.size());
+        assertEquals(new InlineTag("sometag", "some value"), elements.get(0));
+    }
+
+    @Test
+    public void parse_mix_textAndTag() {
+        List<CommentElement> elements = CommentParser.parse("text before {@sometag some value}").getElements();
+        assertEquals(2, elements.size());
+        assertEquals(new CommentText("text before "), elements.get(0));
+        assertEquals(new InlineTag("sometag", "some value"), elements.get(1));
+    }
+
+    @Test
+    public void parse_mix_textAndLink() {
+        List<CommentElement> elements = CommentParser.parse("text before {@link ClassName} text after").getElements();
```

Comment:
Please also add a test for a text like this:

`"Adds the given {@link Action}s to the queue."`

Reply:
Isn't this what this test is about? OR do you mean with no whitespace following the closing brace?

Reply:
Yes. The main use cases usually dont break during a refactor, but these edge cases do.

Reply:
Done.